### PR TITLE
vertex checks

### PR DESCRIPTION
**Pull request by msokoloff1**
* Filter out subclasses to annotation type constructor doesn't break
* Add checks for vertex dataset requirements
* Datasets that fail will show meaningful errors in the UI instead of a cryptic vertex error.

### CODE REVIEW

File: src/jobs/etl/images/bounding_box/bounding_box_etl.py [hunk 1]
```diff
@@ -141,3 +142,6 @@ def main(model_run_id: str, gcs_bucket: str, gcs_key: Optional[str] = None):
     parser.add_argument('--gcs_key', type=str, required=False, default=None)
     args = parser.parse_args()
     main(**vars(args))
+
+
+
```
Comment (gszpak) [inline src/jobs/etl/images/bounding_box/bounding_box_etl.py:147]: nit: remove:)

File: src/jobs/etl/images/bounding_box/bounding_box_etl.py [hunk 2]
```diff
@@ -115,14 +119,11 @@ def bounding_box_etl(lb_client: Client, model_run_id: str, bucket) -> str:
...
         stringified ndjson
     """
-
     labels = get_labels_for_model_run(lb_client, model_run_id)
-    training_data = process_labels_in_threadpool(process_label, labels, bucket)
-
-    if len(training_data) < VERTEX_MIN_TRAINING_EXAMPLES:
-        raise InvalidLabelException("Not enough training examples provided")
+    vertex_labels = process_labels_in_threadpool(process_label, labels, bucket)
+    validate_vertex_dataset(vertex_labels, 'boundingBoxAnnotations')
```
Comment (gszpak) [inline src/jobs/etl/images/bounding_box/bounding_box_etl.py:124]: Don't we want to pass VERTEX_MIN_TRAINING_EXAMPLES here?

File: src/training_lib/training_lib/etl.py [hunk 3]
```diff
@@ -72,7 +75,55 @@ def get_labels_for_model_run(client: labelbox.Client, model_run_id: str, media_t
...
+
+def _get_display_names_for_label(vertex_label, annotation_name):
+    annotation_data = vertex_label[annotation_name]
+    if isinstance(annotation_data, list):
+        for annotation in annotation_data:
+            yield annotation['displayName']
+    else:
+        yield annotation_data['displayName']
+
+def validate_vertex_dataset(vertex_labels, annotation_name, min_labels_per_class = 10, min_classes = None, max_classes = None, min_labels = None):
```
Comment (gszpak) [inline src/training_lib/training_lib/etl.py:98]: Shouldn't min_labels default to `VERTEX_MIN_TRAINING_EXAMPLES` ?

**Review by gszpak — APPROVED** [review]
lgtm! Just a couple of nits

File: src/training_lib/training_lib/etl.py [hunk 4]
```diff
@@ -72,7 +75,55 @@ def get_labels_for_model_run(client: labelbox.Client, model_run_id: str, media_t
...
+    Args:
+        vertex_labels : A list of dictionaries in any vertex compatible format
+        annotation_name: A string used for looking up the displayName for the annotation
+            All vertex annotations have a top level key (annotation_name) that either contains a list of dicts or a dict with a displayName key
+        min_labels_per_class: min number of labels that a class must exist in
+        max_classes: Max total classes in the entire dataset
+        min_labels: Minimum number of labels
+    """
+
+    class_counts = {partition: defaultdict(lambda: 0) for partition in partition_mapping.values()}
```
[the remark this answers is not in the document]
Comment (msokoloff1) [reply src/training_lib/training_lib/etl.py:109]: Done

File: src/training_lib/training_lib/etl.py [hunk 5]
```diff
@@ -72,7 +75,55 @@ def get_labels_for_model_run(client: labelbox.Client, model_run_id: str, media_t
...
+    for vertex_label in vertex_labels:
+        partition = vertex_label['dataItemResourceLabels']['aiplatform.googleapis.com/ml_use']
```
[the remark this answers is not in the document]
Comment (msokoloff1) [reply src/training_lib/training_lib/etl.py:111]: yeah... I'll add typed dicts at some point. They all share this same key.

File: src/jobs/etl/images/bounding_box/bounding_box_etl.py (hunk 2 above)
Comment (msokoloff1) [reply src/jobs/etl/images/bounding_box/bounding_box_etl.py:124]: Technically bounding box doesn't have min examples. It cares about per-class representation.

File: src/training_lib/training_lib/etl.py (hunk 3 above)
Comment (msokoloff1) [reply src/training_lib/training_lib/etl.py:98]: Some of the tasks don't have a min number of examples.

**Pull request merged**